### PR TITLE
fix(hub-common): saving sites guarantees a unique slug and prevents field loss in site data

### PR DESCRIPTION
**Pull request by sonofflynn89**
affects: @esri/hub-common

1. Description:
Issue - https://devtopia.esri.com/dc/hub/issues/7337
Fixes a couple saving issues with the `updateSite` logic

1. [x] Updated meaningful TSDoc to methods including Parameters and Returns, see [Documentation Guide](https://esri.github.io/hub-components/storybook/?path=/story/guides-documentation--page)

1. [x] used semantic commit messages
  
1. [x] PR title follows semantic commit format (**CRITICAL** if the title is not in a semantic format, the release automation will not run!)

1. [ ] updated `peerDependencies` as needed. **CRITICAL** our automated release system can **not** be counted on to update `peerDependencies` so we _must_ do it manually in our PRs when needed. See the [updating peerDependencies](/RELEASE.md#Updating-peerDependencies) section of the release instructions for more details.

### CODE REVIEW

File: packages/common/src/sites/HubSites.ts [hunk 1]
```diff
@@ -309,35 +309,38 @@ export async function updateSite(
   site: IHubSite,
   requestOptions: IHubRequestOptions
 ): Promise<IHubSite> {
+  // verify that the slug is unique, excluding the current site
+  site.slug = await getUniqueSlug(
```
Comment (sonofflynn89) [inline packages/common/src/sites/HubSites.ts:313]: Dave told me that we should use the same slug verification logic as projects, so I introduced it here

**Review by mjuniper — APPROVED** [review]

**Review by mjuniper — APPROVED** [review]

File: packages/common/src/sites/HubSites.ts (hunk 1 above)
Comment (sonofflynn89) [reply packages/common/src/sites/HubSites.ts:313]: @dbouwman, @mjuniper wasn't aware that sites had / needed slugs. Just to confirm before we merge, can we expect _every_ site to have a slug in `item.properties.slug`, or just enterprise sites?

Comment (dbouwman) [reply packages/common/src/sites/HubSites.ts:313]: All "Entity" based items should create a slug, and store it in `item.properties.slug` and a corresponding typekeyword - all done using the same set of helper functions. 

Whether or not anything uses that is a separate question that comes down to how we want urls to work. 

And while it's true that Site's on AGO are unlikely to ever use this property, I'm more inclined to have consistent code for Enterprise and AGO than to have yet-more permutations. :)

**Review by dbouwman — APPROVED** [review]
Thanks @sonofflynn89!

**Pull request merged**